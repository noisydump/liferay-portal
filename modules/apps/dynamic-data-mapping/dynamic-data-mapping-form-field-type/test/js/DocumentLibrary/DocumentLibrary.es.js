/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import DocumentLibrary from '../../../src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es';
import withContextMock from '../__mocks__/withContextMock.es';

let component;
const spritemap = 'icons.svg';

const defaultDocumentLibraryConfig = {
	name: 'textField',
	spritemap,
};

const DocumentLibraryWithContextMock = withContextMock(DocumentLibrary);

describe('Field DocumentLibrary', () => {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('is not readOnly', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			readOnly: false,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a helptext', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			tip: 'Type something',
		});

		expect(component).toMatchSnapshot();
	});

	it('has an id', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			id: 'ID',
		});

		expect(component).toMatchSnapshot();
	});

	it('has a label', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			label: 'label',
		});

		expect(component).toMatchSnapshot();
	});

	it('has a placeholder', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			placeholder: 'Placeholder',
		});

		expect(component).toMatchSnapshot();
	});

	it('is not required', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			required: false,
		});

		expect(component).toMatchSnapshot();
	});

	it('renders Label if showLabel is true', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			label: 'text',
			showLabel: true,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a spritemap', () => {
		component = new DocumentLibraryWithContextMock(
			defaultDocumentLibraryConfig
		);

		expect(component).toMatchSnapshot();
	});

	it('has a value', () => {
		component = new DocumentLibraryWithContextMock({
			...defaultDocumentLibraryConfig,
			value: '{"id":"123"}',
		});

		expect(component).toMatchSnapshot();
	});
});
