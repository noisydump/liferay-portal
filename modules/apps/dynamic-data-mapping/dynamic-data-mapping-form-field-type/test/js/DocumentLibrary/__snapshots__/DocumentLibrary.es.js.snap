// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field DocumentLibrary has a helptext 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div><span class="form-text">Type something</span>
</div>
`;

exports[`Field DocumentLibrary has a label 1`] = `
<div class="form-group" data-field-name="textField">
  <p class="ddm-label">label </p>
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary has a placeholder 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" placeholder="Placeholder" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary has a spritemap 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary has a value 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="{&quot;id&quot;:&quot;123&quot;}">
  </div>
</div>
`;

exports[`Field DocumentLibrary has an id 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input id="ID" name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary is not readOnly 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary is not required 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary renders Label if showLabel is true 1`] = `
<div class="form-group" data-field-name="textField">
  <p class="ddm-label">text </p>
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend"><input aria-label="file" class="field form-control" disabled="" id="textFieldinputFile" type="text" value=""></div>
      <div class="input-group-append input-group-item input-group-item-shrink"><button class="btn btn-secondary select-button" type="button"><span class="lfr-btn-label">select</span></button></div>
    </div><input name="textField" type="hidden" value="">
  </div>
</div>
`;
